### PR TITLE
Added Openai Models fetching

### PR DESCRIPTION
**Pull request by paolocambria**

### CODE REVIEW

**Comment by S1LV3RJ1NX** [comment]
@paolocambria Thank! for the PR. Next time it will be good to add a detailed description of what was the issue, and some details on the resolution.

File: backend/server/routers/internal.py [hunk 1]
```diff
@@ -94,9 +94,16 @@ def get_enabled_models(
                 # OpenAI models
                 url = "https://api.openai.com/v1/models"
                 headers = {"Authorization": f"Bearer {settings.OPENAI_API_KEY}"}
-                response = requests.get(url=url)
+                response = requests.get(url=url, headers=headers)
                 data = response.json()
-                # TODO: Add the models to the enabled_models list
+                                
+                for model in data["data"]:
+                    model_config = {
```
Comment (S1LV3RJ1NX) [inline backend/server/routers/internal.py:101]: It would be good to wrap the model config in `LLMConfig` class.

**Review by S1LV3RJ1NX — CHANGES_REQUESTED** [review]

File: backend/server/routers/internal.py (hunk 1 above)
Comment (chiragjn) [reply backend/server/routers/internal.py:101]: There doesn't  seem to be a model type in the response of this API
How are we supposed to filter for only language models?

Comment (paolocambria) [reply backend/server/routers/internal.py]: This API call returns a list of available models from openai, which at the time of writing are listed below.
With my PR which removes your TODO, I let the user choose a set of OpenAI models in DocsQA, by appending the models fetched by the endpoint. 
I wrapped the model config in LLMConfig class as you suggested.
```
{
    "object": "list",
    "data": [
        {
            "id": "gpt-4-1106-preview",
            "object": "model",
            "created": 1698957206,
            "owned_by": "system"
        },
        {
            "id": "whisper-1",
            "object": "model",
            "created": 1677532384,
            "owned_by": "openai-internal"
        },
        {
            "id": "gpt-4o-2024-05-13",
            "object": "model",
            "created": 1715368132,
            "owned_by": "system"
        },
        {
            "id": "babbage-002",
            "object": "model",
            "created": 1692634615,
            "owned_by": "system"
        },
        {
            "id": "dall-e-2",
            "object": "model",
            "created": 1698798177,
            "owned_by": "system"
        },
        {
            "id": "gpt-3.5-turbo-16k",
            "object": "model",
            "created": 1683758102,
            "owned_by": "openai-internal"
        },
        {
            "id": "tts-1-hd-1106",
            "object": "model",
            "created": 1699053533,
            "owned_by": "system"
        },
        {
            "id": "tts-1-hd",
            "object": "model",
            "created": 1699046015,
            "owned_by": "system"
        },
        {
            "id": "dall-e-3",
            "object": "model",
            "created": 1698785189,
            "owned_by": "system"
        },
        {
            "id": "gpt-3.5-turbo-instruct-0914",
            "object": "model",
            "created": 1694122472,
            "owned_by": "system"
        },
        {
            "id": "gpt-3.5-turbo-instruct",
            "object": "model",
            "created": 1692901427,
            "owned_by": "system"
        },
        {
            "id": "text-embedding-3-small",
            "object": "model",
            "created": 1705948997,
            "owned_by": "system"
        },
        {
            "id": "gpt-4-turbo-2024-04-09",
            "object": "model",
            "created": 1712601677,
            "owned_by": "system"
        },
        {
            "id": "tts-1",
            "object": "model",
            "created": 1681940951,
            "owned_by": "openai-internal"
        },
        {
            "id": "gpt-4-turbo",
            "object": "model",
            "created": 1712361441,
            "owned_by": "system"
        },
        {
            "id": "text-embedding-3-large",
            "object": "model",
            "created": 1705953180,
            "owned_by": "system"
        },
        {
            "id": "gpt-3.5-turbo-1106",
            "object": "model",
            "created": 1698959748,
            "owned_by": "system"
        },
        {
            "id": "gpt-4-0125-preview",
            "object": "model",
            "created": 1706037612,
            "owned_by": "system"
        },
        {
            "id": "gpt-3.5-turbo-0125",
            "object": "model",
            "created": 1706048358,
            "owned_by": "system"
        },
        {
            "id": "gpt-4o",
            "object": "model",
            "created": 1715367049,
            "owned_by": "system"
        },
        {
            "id": "gpt-3.5-turbo",
            "object": "model",
            "created": 1677610602,
            "owned_by": "openai"
        },
        {
            "id": "gpt-3.5-turbo-0301",
            "object": "model",
            "created": 1677649963,
            "owned_by": "openai"
        },
        {
            "id": "gpt-4-turbo-preview",
            "object": "model",
            "created": 1706037777,
            "owned_by": "system"
        },
        {
            "id": "tts-1-1106",
            "object": "model",
            "created": 1699053241,
            "owned_by": "system"
        },
        {
            "id": "gpt-3.5-turbo-16k-0613",
            "object": "model",
            "created": 1685474247,
            "owned_by": "openai"
        },
        {
            "id": "gpt-3.5-turbo-0613",
            "object": "model",
            "created": 1686587434,
            "owned_by": "openai"
        },
        {
            "id": "gpt-4",
            "object": "model",
            "created": 1687882411,
            "owned_by": "openai"
        },
        {
            "id": "text-embedding-ada-002",
            "object": "model",
            "created": 1671217299,
            "owned_by": "openai-internal"
        },
        {
            "id": "gpt-4-1106-vision-preview",
            "object": "model",
            "created": 1711473033,
            "owned_by": "system"
        },
        {
            "id": "davinci-002",
            "object": "model",
            "created": 1692634301,
            "owned_by": "system"
        },
        {
            "id": "gpt-4-0613",
            "object": "model",
            "created": 1686588896,
            "owned_by": "openai"
        },
        {
            "id": "gpt-4-vision-preview",
            "object": "model",
            "created": 1698894917,
            "owned_by": "system"
        }
    ]
}
```

Comment (S1LV3RJ1NX) [reply backend/server/routers/internal.py]: what the earlier meant was, we have model types like chat, embedding and completion. Can we have segregation based on that? Either after we get the response or if openai provides a separate API for them. You can have a look at `if settings.TFY_API_KEY:` code block

**Review by innoavator — APPROVED** [review]

**Review by innoavator — APPROVED** [review]

**Comment by S1LV3RJ1NX** [comment]
@paolocambria , though the PR is approved, suggested changes are required before we merge.
Thanks!

**Comment by S1LV3RJ1NX** [comment]
@paolocambria , can you create a master list out of model id's like if model contains `gpt-` it's a chat model, and a concrete list of embedding models if present?

**Comment by paolocambria** [comment]
Sure, I will push that soon

**Comment by paolocambria** [comment]
I pushed, but I thought it was possible to add also a "provider" extracted by "owned_by".
I didn't do that because system is not informative.

**Comment by paolocambria** [comment]
This is not ready for merge. As far as I see, I should adapt the call to the embedding service too, because it's not in a format openAI accepts.

**Comment by innoavator** [comment]
@paolocambria : I am pushing out another change to make providing these models in a better way using a config file. I would suggest holding on to this PR for a couple of days - that change will solve the problem we are trying to solve here.

**Comment by paolocambria** [comment]
@innoavator it's ok I pushed a working version

**Comment by innoavator** [comment]
This PR will solve this issue generically: https://github.com/truefoundry/cognita/pull/222 - so we will not need this PR once that PR is merged:

**Pull request closed without merging**